src/main/java/ninjabrainbot/model/datastate/endereye/CoordinateInputSource.java

```diff
@@ -53,6 +60,13 @@ private void parseF3C(String f3c) {
 		if (f != null) {
 			whenNewFossilInputted.setAndAlwaysNotifySubscribers(f);
 		}
+
+		if (preferences.oneDotTwentyPlusAA.get() && preferences.allAdvancements.get()) {
```

Comment:
Few comments about this:

- AA preferences should not matter here. This class's only purpose is to provide the coordinates the user inputs, it should not care about the AA preferences. setAndAlwaysNotifySubscribers should always be called whenever a new F3+I is detected, but the subscriber should then determine whether or not to do something (or ignore the F3+I) based on the AA preferences.
- Parsing should not be done in the StructurePosition class, CoordinateInputSource is a general class for providing what the user inputs, its too specific to have StructurePosition code here. Instead create a class similar to F3CData, and do F3IData.tryParseF3IString(f3c);
- I do not like the name "GeneralLocation" because it refers to the specific usage in the new AA mode. I think a better name is just F3ILocation. Rename this everywhere, e.g. GeneralLocationInputHandler -> F3ILocationInputHandler.

Reply:
Btw you can keep the name "GeneralLocation" in the places where it specifically referrs to the general location used in AA mode, such as in IAllAdvancementsDataState